Add assign-oauth-image gatekeeper policy

IsaiahStapleton opened the pull request:
This allows the rhods oauth container to pull oauth image from internal registry rather than externally. Relevant issue: https://github.com/nerc-project/operations/issues/506

larsks commented:
In your commit message, you write `relevant commit:` where I think you meant `relevant issue:`.

larsks commented on policy/overlays/nerc-ocp-prod/internal-oauth.yaml:
I think I would like this renamed to something like `use-internal-oauth-proxy-image.yaml`. I realize that's a bit of a mouthful, but: the reason I was confused by the filename earlier was because `internal-oauth` sounds like something related to our auth configuration.

IsaiahStapleton committed:
Add assign-oauth-image gatekeeper policy

This allows the rhods oauth container to pull oauth image from internal registry rather than externally.

Relevant issue: nerc-project/operations#506

Signed-off-by: Isaiah Stapleton <istaplet@redhat.com>
